### ISSUE TITLE
[Bug]: Pimcore invalidates all thumbnails as soon as you just press Save on the asset

### ISSUE THREAD

**Issue by dkarlovi** [pimcore/pimcore#15273]
### Pimcore version

10.5, 10.6

### Steps to reproduce

Have say 5 thumbnail configs, auto-formats + 1x, 2x, you come out to about 45 thumbnails per image.

If I go to Asset edit and just press Save (even without doing any changes), suddenly all of those thumbnails are considered out of date and will need to be regenerated. If you say import alt metadata to 10k images, **suddenly you have 500k thumbnails to generate** even though they should all still be up to date.

### Expected / actual behavior

Thumbnails should be considered **out of date if bytes in the thumbnail file would have changed** if we regenerated it compared to what we have now. In any other case, the thumbnail should be seen as up to date.

Edit: see here for more info why this is bad:
https://github.com/pimcore/pimcore/issues/15231#issuecomment-1572012136

**Comment by dkarlovi** [comment]
This is partly due to a long running issue the modified date always changes when you save(), even if no actual changes were made, but here it's really expensive.

**Comment by brusch** [comment]
@dkarlovi actually this only happens when data has changed, see: https://github.com/pimcore/pimcore/blob/d982fa8d96dc775e49a04017cfa657882cedf148/models/Asset.php#L1600

**Comment by dkarlovi** [comment]
@brusch press Save on an asset and then rerun `pimcore:thumb:image --id <ID>`

**Comment by dkarlovi** [comment]
It's because of this:

https://github.com/pimcore/pimcore/blob/9b1d3eb47626ed5386448d396c7100d8ebefbd23/models/Asset/Image/Thumbnail/Processor.php#L186

**Comment by dkarlovi** [comment]
But even if we actually do edit the asset, the thumbnail shouldn't automatically get invalidated just because we updated a random metadata property. If the bytes in the thumbnail file are still valid (as in, we would get the same bytes if we regenerated the thumbnail now), it shouldn't get invalidated. This is probably a big ask for Pimcore 10.x, but should work like that for Pimcore 11. This "no change, still invalidated" should be fixed in Pimcore 10 IMO.
